lib/validation.rb

```diff
@@ -325,4 +325,23 @@ def self.validate_rfc3339_datetime_str(datetime_str, param = "time")
     msg = "\"#{datetime_str}\" is not a valid date for \"#{param}\"."
     fail ValidationFailed.new({param => msg})
   end
+
+  def self.validate_from_option_tree(option_tree, option_parents, params)
+    params.sort_by { |k, _| option_parents[k].count }.each do |key, value|
+      parents = option_parents[key]
+      subtree = option_tree
+
+      parents.each do |parent|
+        parent_value = params[parent]
+        subtree = subtree[parent][parent_value]
+      end
+
+      if subtree[key][value].nil?
+        display_key = key.tr("_", " ")
+        value = value.is_a?(Location) ? value.display_name : value.to_s
+        available_options = subtree[key].keys.map! { it.is_a?(Location) ? it.display_name : it.to_s }.join(", ")
```

Comment:
Not sure how often we do this, but if it is in more than these 2 places, it would be good to have a method for the `display_name` or `to_s` logic to DRY things up.